lib/matplotlib/axes/_axes.py

```diff
@@ -6664,7 +6699,7 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
             # so that each histogram uses the same bins
             m, bins = np.histogram(x[i], bins, weights=w[i], **hist_kwargs)
             m = m.astype(float)  # causes problems later if it's an int
-            if mlast is None:
+            if stacked and mlast is None:
```

Comment:
is that fixing a separate bug (or not?)

Reply:
mirco-optomization, we only care if this is non-None if we are stacking.

Reply:
then move the condition below into this one
```
if stacked:
    if mlast is None: mlast = ...
    m += mlast
    mlast[:] = m
```
or even better
```
mlast = 0  # instead of None
if stacked:
    m += mlast
    mlast = m.copy()
```
seems easier to follow (sure, you don't get to reuse the same buffer, but I doubt that's relevant).